src/main/java/de/qabel/desktop/ui/accounting/AccountingController.java

```diff
@@ -79,87 +72,148 @@ public void addIdentity() {
-		}
-	}
-
 	@FXML
-	protected void handleImportIdentityButtonAction(ActionEvent event) {
+	protected void handleImportIdentityButtonAction(ActionEvent event) throws URISyntaxException, QblDropInvalidURL {
 
 		FileChooser chooser = new FileChooser();
 		chooser.setTitle(resourceBundle.getString("downloadFolder"));
 		File file = chooser.showOpenDialog(identityList.getScene().getWindow());
 		try {
 			importIdentity(file);
+			loadIdentities();
 		} catch (IOException | PersistenceException e) {
 			e.printStackTrace();
 		}
 	}
 
-	void importIdentity(File file) throws IOException, PersistenceException {
-		String content = readFile(file);
-		Identity i = gson.fromJson(content, Identity.class);
-		identityRepository.save(i);
-	}
-
 	@FXML
 	protected void handleExportIdentityButtonAction(ActionEvent event) {
-		DirectoryChooser chooser = new DirectoryChooser();
-		chooser.setTitle(resourceBundle.getString("downloadFolder"));
-		File dir = chooser.showDialog(identityList.getScene().getWindow());
+
 		Identity i = clientConfiguration.getSelectedIdentity();
+		File file = createSaveFileChooser(i.getAlias() + "_Identity.json");
 		try {
-			exportIdentity(i, dir);
+
+			exportIdentity(i, file);
 			loadIdentities();
 		} catch (IOException | QblStorageException e) {
 			e.printStackTrace();
 		}
 	}
 
-	void exportIdentity(Identity i, File dir) throws IOException, QblStorageException {
+	@FXML
+	protected void handleExportContactButtonAction(ActionEvent event) {
+		Identity i = clientConfiguration.getSelectedIdentity();
+		File file = createSaveFileChooser(i.getAlias() + "_Contact.json");
+		try {
+			exportContact(i, file);
+		} catch (IOException | QblStorageException e) {
+			e.printStackTrace();
+		}
+	}
 
-		String json = gson.toJson(i);
-		InputStream stream = new ByteArrayInputStream(json.getBytes(StandardCharsets.UTF_8));
-		byte[] buffer = new byte[stream.available()];
-		stream.read(buffer);
+	protected void addIdentityWithAlias(String alias) {
+		Identity identity = identityBuilderFactory.factory().withAlias(alias).build();
+		try {
+			identityRepository.save(identity);
+		} catch (PersistenceException e) {
+			alert("Failed to save new identity", e);
+		}
+		loadIdentities();
+		if (clientConfiguration.getSelectedIdentity() == null) {
+			clientConfiguration.selectIdentity(identity);
+		}
+	}
+
+	void importIdentity(File file) throws IOException, PersistenceException, URISyntaxException, QblDropInvalidURL {
+		String content = readFile(file);
+		GsonIdentity gi = gson.fromJson(content, GsonIdentity.class);
+		Identity i = gsonIdentityToIdentiy(gi);
+		identityRepository.save(i);
+	}
+
+	void exportIdentity(Identity i, File file) throws IOException, QblStorageException {
+		GsonIdentity gi = createGsonIdentity(i);
+		String json = gson.toJson(gi);
+		writeJsonInFile(json, file);
+	}
+
+	void exportContact(Identity i, File file) throws IOException, QblStorageException {
+		GsonContact gc = createGsonFromEntity(i);
```

Comment:
The naming is misleading. Looking at `createGsonFromEntity` the question raises what `createGsonIdentity` is for until realizing that `createGsonFromEntity` is just a `createGsonContact`. I would still prefer some place outside the controller like `GsonContact.fromEntity(Entity e)` and `GsonIdentity.fromIdentity(Identity i)`